src/adapter/src/catalog/migrate.rs

```diff
@@ -387,70 +385,71 @@ fn ast_rewrite_create_materialized_view_refresh_options_0_82_0(
     Ok(())
 }
 
-/// Rewrite all non-`pg_catalog` system types to have the correct schema.
-fn ast_rewrite_linked_cluster_sizes<'a>(
+/// To deprecate the notion of linked clusters, convert all linked clusters into
+/// unlinked clusters.
+fn ast_unlink_all_linked_clusters<'a>(
+    conn_catalog: &'a &ConnCatalog<'_>,
+    id: GlobalId,
     stmt: &'a mut Statement<Raw>,
-    state: &'a std::borrow::Cow<'a, CatalogState>,
 ) {
     use mz_sql::ast::visit_mut::VisitMut;
 
     struct Rewriter<'a> {
-        state: &'a std::borrow::Cow<'a, CatalogState>,
+        conn_catalog: &'a ConnCatalog<'a>,
+        id: GlobalId,
     }
 
-    impl<'ast> VisitMut<'ast, Raw> for Rewriter<'_> {
+    impl<'ast, 'a> VisitMut<'ast, Raw> for Rewriter<'a> {
         fn visit_create_source_statement_mut(
             &mut self,
             node: &'ast mut mz_sql_parser::ast::CreateSourceStatement<Raw>,
         ) {
-            // If no specified cluster and no specified size, then this is an
-            // "old"-style source that used a linked cluster of the default
-            // size.
-            if node.in_cluster.is_none()
-                && !node
-                    .with_options
-                    .iter()
-                    .any(|o| o.name == mz_sql_parser::ast::CreateSourceOptionName::Size)
-            {
-                node.with_options
-                    .push(mz_sql_parser::ast::CreateSourceOption {
-                        name: mz_sql_parser::ast::CreateSourceOptionName::Size,
-                        value: Some(mz_sql_parser::ast::WithOptionValue::Value(
-                            mz_sql_parser::ast::Value::String(
-                                self.state.default_linked_cluster_size(),
-                            ),
-                        )),
-                    })
+            node.with_options
+                .retain(|o| o.name != mz_sql_parser::ast::CreateSourceOptionName::Size);
+
+            if let Some(cluster) = self.conn_catalog.state.get_linked_cluster(self.id) {
+                mz_ore::soft_assert_or_log!(
```

Comment:
Why is it correct for this to proceed instead of panic in prod?

Reply:
If this were the case in prod, we would need to fix it, and the fix would just be doing exactly what we're doing below.

My hope was just that if this invariant got broken we'd have some signal and could reconsider the approach taken in this PR. I don't think this is a high-value assert or anything--just being defensive.